pkg/template/generator/expressionvalue.go

```diff
@@ -28,15 +28,16 @@ type ExpressionValueGenerator struct {
 }
 
 const (
-	Alphabet = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
-	Numerals = "0123456789"
-	ASCII    = Alphabet + Numerals + "~!@#$%^&*()-_+={}[]\\|<,>.?/\"';:`"
+	Alphabet    = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	Numerals    = "0123456789"
+	NonAlphabet = "~!@#$%^&*()-_+={}[]\\|<,>.?/\"';:`"
+	ASCII       = Alphabet + Numerals + NonAlphabet
 )
 
 var (
 	rangeExp      = regexp.MustCompile(`([\\]?[a-zA-Z0-9]\-?[a-zA-Z0-9]?)`)
 	generatorsExp = regexp.MustCompile(`\[([a-zA-Z0-9\-\\]+)\](\{([0-9]+)\})`)
-	expressionExp = regexp.MustCompile(`\[(\\w|\\d|\\a)|([a-zA-Z0-9]\-[a-zA-Z0-9])+\]`)
+	expressionExp = regexp.MustCompile(`\[(\\w|\\d|\\a|\\S)|([a-zA-Z0-9]\-[a-zA-Z0-9])+\]`)
```

Comment:
what is this doing: ([a-zA-Z0-9]\-[a-zA-Z0-9])+
?
shouldn't it just match [\w\d\S]+ ?